include/boost/geometry/algorithms/simplify.hpp

```diff
@@ -116,34 +117,174 @@ struct simplify_range
                     range, geometry::range::back_inserter(out), max_distance, strategy
                 );
         }
+
+        // Verify the two remaining points are equal. If so, remove one of them.
+        // This can cause the output being under the minimum size
+        if (boost::size(out) == 2
+            && geometry::equals(geometry::range::front(out),
+                    geometry::range::back(out)))
```

Comment:
This is the second time this condition is present in the code. This check could be implemented as a free function.